GitHub release text must contain content of release info files

ngoan1608 opened the pull request:
Hi team,

This PR implements the issue #324.

I have extended current `release_info` tool to generate the `release_changelog.html` file which is used as Github Release content.
I used flat format for Github Release content instead of table of release_info html file. I have tried with table format before but my feeling is not good.

Referent Github Release content: https://github.com/ngoan1608/RobotFramework_AIO/releases/tag/rel%2Faio%2F0.13.0.6

(Referent content that uses table format: https://github.com/ngoan1608/RobotFramework_AIO/releases/tag/rel%2Faio%2F0.13.0.5)

Please review and give feeling about it.

Thank you,
Ngoan

ngoan1608 committed:
update release_info tool to generate changelog file for Release content

ngoan1608 committed:
update build pipeline to add Release content

ngoan1608 committed:
avoid wraptext in changelog html content

ngoan1608 committed:
use flat format for changelog content rather than a table

ngoan1608 committed:
fix merge conflict

test-fullautomation reviewed (changes requested):
Hi Ngoan,

very nice!
I actually prefer the table format, but it requires a small change.
The repository name need to be in top left of the corresponding cell instead of vertical center. This makes it much better readable.
The table format with the alternating cell background brings a better stucture into the text than the plain text with numbers.
Please change accordingly.

Thank you,
 Thomas

ngoan1608 committed:
Merge branch 'develop' into ntd1hc/feat/324-github-release-text-must-contain-content-of-release_info-files

ngoan1608 committed:
change changelog format to use table format

ngoan1608 commented:
Hi Thomas,

I have changed to use table as your proposal.
I have combined all changes of single component into single row instead of multiple rows in release_info file, The numeric column is also removed.

It will look as below:

Please give your feeling about it and let me know in case of any change.
P/s: referent Github Release content https://github.com/ngoan1608/RobotFramework_AIO/releases/tag/rel%2Faio%2F0.13.1.0

Thank you,
Ngoan

<h3><font face="Arial" color="#242424">Changelog</font></h3>
<table width="100%" border="1" cellspacing="0" cellpadding="0" frame="box" rules="all" align="left">

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>PrometheusInterface</h3><p>Initial introduction of keyword library <code><font font-family="courier new" color="navy" size="+1">PrometheusInterface</font></code> to enable monitoring of <strong>Robot Framework</strong> test data.</p> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>PyTestLog2DB</h3><ul> <li>Introduce new feature to switch the access interface to TestResult's database <ul> <li><p>Add supported new interface to database <strong>REST API Access</strong>.</p> <p>This interface helps to import test result data to the secured server which is not allowed to interact directly with the database.</p></li> <li><p>Add new optional argument <code><font font-family="courier new" color="navy" size="+1">--interface</font></code> which allow to switch between available interfaces (currently only direct access). Its value can be:</p> <ul> <li><code><font font-family="courier new" color="navy" size="+1">db</font></code> (default value): use direct access to TestResult's database.</li> <li><code><font font-family="courier new" color="navy" size="+1">rest</font></code>: use REST API Access to TestResult's database (new access)</li> </ul></li> <li><p>This change is transparent to end user.</p></li> </ul></li> </ul> <ul> <li>Add new optional argument <code><font font-family="courier new" color="navy" size="+1">--testrunurl</font></code> to specify a link to test execution job, such as Jenkins or Gitlab CI/CD</li> <li>Fix issue with parsing timestamps from pytest version <code><font font-family="courier new" color="navy" size="+1">8.3.1</font></code> result files</li> </ul> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>RobotFramework AIO Documentation</h3><p>Added documentation of <strong>TestResultDBAccess</strong> package</p> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>RobotFramework AIO Package</h3><p><strong>Python Packages</strong></p> <ul> <li>Integrated <strong>robotframework-prometheus</strong> library</li> <li>Integrated <strong>prometheus_client</strong> library</li> <li>Integrated <strong>pika</strong> library</li> </ul> <p><strong>Python Package Index (PyPI)</strong></p> <ul> <li>Publish Robotframework-prometheus package to <a href="https://pypi.org/user/ThPoll">PyPI</a></li> </ul> <p><strong>Environment/Installation</strong></p> <ul> <li>Debian package: <ul> <li><p>Enhanced <strong>initRobotFrameworkAIO.sh</strong>, support to setup environment for non-root user</p></li> <li><p>Introduce new user group <strong>robot-aio</strong> which is granted required permissions to use <em>robotframework-aio</em> package</p> <p>An user (BOT or integrator account) which have non-root permission can also be setup to use the installed robotframework-aio package on test machine as following steps:</p> <ul> <li><p>Add BOT or Integrator account to robot-aio group by execute below command with root user or add it as a <strong>RUN</strong> step of Docker file</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">usermod -a -G robot-aio $BOT_USER</font></code></pre></li> <li><p>Login as BOT or Integrator account</p></li> <li><p>Execute <strong>initRobotFrameworkAIO.sh</strong> script to setup enviroment or add it as <strong>CMD</strong> step when starting Docker container</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">initRobotFrameworkAIO.sh</font></code></pre></li> <li><p>Now, execute your Robotframework test case/suite. E.g</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">robot ~/RobotTest/testcases/HelloWorld.robot</font></code></pre></li> </ul></li> <li><p>Added executable <strong>appium</strong> script to start the Appium server in command line (useful for Docker container)</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">$RobotDevtools/appium</font></code></pre></li> </ul></li> </ul> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>RobotLog2DB</h3><p>Introduce new feature to switch the access interface to TestResult's database</p> <ul> <li><p>Add supported new interface to database <strong>REST API Access</strong>.</p> <p>This interface helps to import test result data to the secured server which is not allowed to interact directly with the database.</p></li> <li><p>Add new optional argument <code><font font-family="courier new" color="navy" size="+1">--interface</font></code> which allow to switch between available interfaces (currently only direct access). Its value can be:</p> <ul> <li><code><font font-family="courier new" color="navy" size="+1">db</font></code> (default value): use direct access to TestResult's database.</li> <li><code><font font-family="courier new" color="navy" size="+1">rest</font></code>: use REST API Access to TestResult's database (new access)</li> </ul></li> <li><p>This change is transparent to end user.</p></li> </ul> <p>Fixed issue when importing test case with <strong>SKIP</strong> status</p> <p>Add new optional argument <code><font font-family="courier new" color="navy" size="+1">--testrunurl</font></code> to specify a link to test execution job, such as Jenkins or Gitlab CI/CD</p> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>RobotLog2RQM</h3><p>Fixed issue when importing test case with <strong>SKIP</strong> status</p> <p><strong>New feature of testsuite</strong></p> <ul> <li><p>New optional argument: <em>--testsuite &lt;TESTSUITEID&gt;</em></p> <p>The new --testsuite argument allows you to specify how test results are imported into IBM Rational Quality Manager (RQM). You can now import test results into an existing test suite or create a new one if needed.</p></li> <li><p>Usage:</p> <ul> <li><p>Import to existing test suite:</p> <p>To import test results into an existing test suite, simply provide the test suite ID with the <em>--testsuite</em> argument.</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM resultxmlfile &lt;host&gt; &lt;project&gt; &lt;user&gt; &lt;password&gt; &lt;testplan&gt; --testsuite &lt;EXISTING_TESTSUITE_ID&gt;</font></code></pre> <p>This command will import the test results into the test suite specified by &lt;EXISTING_TESTSUITE_ID&gt;.</p></li> <li><p>Create a new test suite:</p> <p>If you want to create a new test suite and import the test results into it, use the value <strong>new</strong> for the <em>--testsuite</em> argument.</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM resultxmlfile &lt;host&gt; &lt;project&gt; &lt;user&gt; &lt;password&gt; &lt;testplan&gt; --testsuite new</font></code></pre> <p>This command will create a new test suite in RQM and then import the test results into the newly created test suite.</p></li> </ul></li> <li><p>Example scenarios:</p> <ul> <li><p>Import to existing test suite with id <strong>1234</strong></p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM robot_result.xml https://rqm-host.com My_RQM_Project testuser testpass 123 --testsuite 1234</font></code></pre></li> <li><p>Create and import to new test suite</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM robot_result.xml https://rqm-host.com My_RQM_Project testuser testpass 123 --testsuite new</font></code></pre></li> </ul></li> </ul> <p><strong>New feature to support project which enable Configuration Management (CM)</strong></p> <ul> <li><p>New optional arguments:</p> <p>When Configuration Management (CM) is enabled for the project area, it is necessary to specify the project stream or baseline when executing import tool to interact with the proper stream/baseline resources.</p> <p>The following arguments are used to provide this information:</p> <ul> <li><em>--stream &lt;STREAM&gt;</em>: A project stream represents a branch of work within a project area.</li> <li><em>--baseline &lt;BASELINE&gt;</em>: A baseline captures the state of project artifacts at a particular point in time.</li> </ul></li> <li><p>Example usages:</p> <ul> <li><p>Using stream <code><font font-family="courier new" color="navy" size="+1">Develop_stream</font></code> for importing test execution results:</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM output.xml https://sample-rqm-host.com ROBFW-AIO test_user test_pw 720 --stream &#39;Develop_stream&#39;</font></code></pre></li> <li><p>Using baseline <code><font font-family="courier new" color="navy" size="+1">Baseline_v1</font></code> for importing test execution results:</p> <pre class=""><code><font font-family="courier new" color="navy" size="+1">python -m RobotLog2RQM output.xml https://sample-rqm-host.com ROBFW-AIO test_user test_pw 720 --baseline &#39;Baseline_v1&#39;</font></code></pre></li> </ul></li> </ul> </font></td>
</tr>

<tr>
   <td bgcolor="#F5F5F5" width="100%" align="left" style="padding:6pt 6pt 6pt 6pt"><font face="Arial" color="#000000" size="-1"><h3>TestResultDBAccess</h3><p>Support added for new interface <strong>REST API</strong></p> </font></td>
</tr>

test-fullautomation reviewed (approved):
Hi Ngoan,
looks good to me!
Thank you,
 Thomas
